src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,119 @@
+package duke;
+
+import java.io.IOException;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+
+import java.util.ArrayList;
+
+import java.util.List;
+
+/**
+ * Class that deals with loading and saving of file.
+ */
+public class Storage {
+    private Path folderPath;
+    private Path filePath;
+
+    public Storage() {
+        folderPath = Paths.get("data");
+        filePath = Paths.get("data/duke.txt");
+    }
+
+    /**
+     * Gets the file path of the file if it already exists and create the file if not.
+     *
+     * @return Path of the file.
+     * @throw IOException.
+     */
+    public Path getFilePath() throws IOException {
+        if (!Files.exists(folderPath)) {
+            folderPath = Files.createDirectory(folderPath);
+            filePath = Files.createFile(filePath);
+            return filePath;
+        } else {
+            if (!Files.exists(filePath)) {
+                filePath = Files.createFile(filePath);
+            }
+            return filePath;
+        }
+    }
+
+    /**
+     * Loads the file and produce it as list.
+     *
+     * @param p Path of the file.
+     * @return List of task.
+     * @throw IOException.
+     */
+    public List<Task> load(Path p) throws IOException {
+        String currContent;
+        String[] parts;
+        ArrayList<Task> taskList = new ArrayList<>();
+        List<String> fileContents = Files.readAllLines(p);
+        for (int i = 0; i< fileContents.size(); i++) {
```

Comment:
Should there be a whitespace before the operator?
```suggestion
        for (int i = 0; i < fileContents.size(); i++) {
```